packages/ui/src/theme/__tests__/createTheme.test.ts

```diff
@@ -73,6 +73,58 @@ describe('@aws-amplify/ui', () => {
       });
     });
 
+    describe('deprecated state tokens', () => {
+      /*
+        Here we are test that deprecated state tokens are being handled correctly and that is any tokens marked with the deprecatedStateToken
+        flag in the base theme should copy over any values found on the deprecated state token to the conforming version (i.e. copy values from hover to _hover)
+        the exception being if the passed in theme already contains a value in the corresponding design token.
+      */
+      it('should add in conforming versions of the deprecated theme tokens', () => {
+        const newTheme = createTheme({
+          name: 'test-theme',
+          tokens: {
+            components: {
+              pagination: {
+                button: {
+                  hover: {
+                    backgroundColor: { value: '#ff9900' },
+                  },
+                },
+              },
+            },
+          },
+        });
+        const { tokens } = newTheme;
+        expect(
+          tokens.components.pagination.button._hover.backgroundColor.value
+        ).toEqual('#ff9900');
+      });
+
+      it('should keep the passed in _hover value', () => {
+        const newTheme = createTheme({
+          name: 'test-theme',
+          tokens: {
+            components: {
+              pagination: {
+                button: {
+                  hover: {
+                    backgroundColor: { value: '#ff9900' },
+                  },
+                  _hover: {
+                    backgroundColor: { value: '#0099ff' },
+                  },
+                },
+              },
+            },
+          },
+        });
+        const { tokens } = newTheme;
+        expect(
+          tokens.components.pagination.button._hover.backgroundColor.value
```

Comment:
What would we expect the value for `hover` to be, without the underscore?

Reply:
Here we would expect the `hover` without the underscore to be `#ff9900`.  I'll update the test to demonstrate this.  What this test is attempting to demonstrate is that if both `hover` and `_hover` is passed into create theme then `_hover` will not be overwritten by `hover`.  The way the logic works out is that create theme will now filter out the tokens marked with `deprecatedStateToken` and so what we are doing is copying the values over from the deprecatedStateTokens to the `_` version.  So if only `hover` exists in the passed in theme it will copy the value over to `_hover`, if only `_hover` exists then it will do nothing, and what this test demonstrates is that if both `hover` and `_hover` are passed in then `_hover` is the value being used.